server/document-watch.worker.js

```diff
@@ -21,41 +33,24 @@ function postDocumentInfo(filePath, changeType) {
     if (!document) {
       return;
     }
-
-    // We check that the metadata (through `document.url`)
-    // matches the filePath by using `Document.urlToFolderPath`.
-    // This would prevent the document from being added in the first place
-    // if the filePath doesn't map correctly to the URL, but in reverse.
-    if (!filePath.includes(Document.urlToFolderPath(document.url))) {
-      console.warn(
-        `The slug of ${filePath} doesn't match the folder is located in.`
-      );
-      return;
-    }
-
     postEvent(changeType, { filePath, document });
   } catch (e) {
     console.error(`Error while adding document ${filePath} to index:`, e);
   }
 }
 
-const SEARCH_PATTERN = path.join(CONTENT_ROOT, "en-us", "**", "*.html");
+const SEARCH_ROOT = path.join(CONTENT_ROOT, "en-us");
 
-const label = "Populate search-index with glob";
+const label = "Populate search-index";
 console.time(label);
 let count = 0;
-glob.sync(SEARCH_PATTERN).forEach((filePath) => {
-  // Note! On Windows, `glob.sync()` will return URLs that always use
-  // the `/` character (POSIX STYLE, https://www.npmjs.com/package/glob#windows)
-  // But the `Document.urlToFolderPath()` function will respect Windows
-  // notation and use `\` characters.
-  // Now, when we use it, it'll work as if `glob.sync()` and returned paths
-  // in a fashion that is expected in the OS.
-  postDocumentInfo(filePath.split("/").join(path.sep), "added");
-  count++;
-});
+for (const filePath of walker(SEARCH_ROOT)) {
+  const basename = path.basename(filePath);
+  if (basename === "index.html" || basename === "index.md") {
+    postDocumentInfo(filePath, "added");
+    count++;
+  }
+}
```

Comment:
Doing this dead-simple walker, instead of the `glob.sync` is actually marginally faster. 
Now, if it also doesn't have to do that `if (!filePath.includes(Document.urlToFolderPath(document.url))) {` check, the whole loop is actually faster. 

I extracted the relevant code in a separate file and ran it 10 times. It's actually 33% faster. ~2.2 seconds instead of ~3.3 seconds. 
It might seem silly but the `document-watch.worker.js` is invoked every single time you run `yarn dev` or if the `nodemon` restarts the server. So it's a welcome change.

Reply:
That speed-up is due to 3 different things I don't know which matters the most.

- faster than `glob.sync()` which is a more powerful thing that me simple recursive walker
- not having to do the comparison of the folder name with what the it became thanks to the slug
- not having to do that `filePath.split("/").join(path.sep)` trick because `glob.sync()` yields POSIX style paths on Windows.